### ISSUE TITLE
🚀 available new version - many improvements 🚀

### ISSUE THREAD

**Issue by hex13** [hex13/atom-lupa#12]
I'm working on a new version of package. 

I'm practically rewriting big parts of this (I had to do this to improve architecture/code structure of the project). I'm also improving many things in the interface. 

Probably I'm gonna release next version this weekend.

**Comment by hex13** [comment]
New version will be released tomorrow. (I don't want to ship now because there are still some features that are in the current version and I haven't reimplemented them in the new version)

For now:
- [x] added the handle for resize side panel 
- [x] improved preview 
- [X] fixed weird scroll behavior (caused by Atom bug)
- [X] done some other improvements

There are still some thing to do though:
- [x] restoring full project indexing 
- [x] restoring global search (I don't remember if I documented that but on the sidepanel you can trigger global search by beginning phrase with two dots e.g.

```
..somename
```
- [x] restoring clicking imports/requires (go to file)
- [ ] restoring project explorer (not essential for now*)
- [X] colors should be more blended with the user theme (now the package works best with One Dark theme, but I want to make it theme-agnostic.  

From all of this only theming/color thing would be a little work more. I could just replace hardcoded values like `#aaa` to Atom variables in Less  like `@text-color` and it would be more integrated with Atom. Yeah. But this is not that simple. I don't want to couple styling to Atom because I'm planning to decuple this whole plugin (and make e.g. VSCode version or Chrome Extension from it etc.).

So I think how to decouple styles/CSS class names and inject them later into React components (I'm gonna look out for these cool libraries like Radium or CSSModules, maybe there will be an answer).

*project explorer was still in beta anyway and was not implemented or designed very good.

**Comment by hex13** [comment]
Now it's more fitted to user theme. Effect: 
![image](https://cloud.githubusercontent.com/assets/3883359/15302152/bf3e56e2-1bb1-11e6-9af4-91a3fc05d783.png)

It works with light themes too:
![image](https://cloud.githubusercontent.com/assets/3883359/15302188/f2d1f9aa-1bb1-11e6-8d0c-0432f6b7458d.png)

Styling wasn't that hard actually.
1. I made additional CSS classes just for colors and put Atom variables there, i.e.
   
   ``` less
   .lupa-text {
       color: @text-color;
   }
   .lupa-link {
       color: @text-color-info;
   }
   ```
2. I made some simple helper function
   
   ``` javascript
   function getClass(entity, initialClass) {
       // TODO this is naive. `entity.type` and classes uses in Atom don't always match
       // but it works with function and with class
       return ['entity', 'name', entity.type].concat(initialClass).join(' ');
   }
   ```
3. and created simple wrapper with `source` and `js` class. Into this wrapper I inject entity classes:
   
   ``` javascript
           return <span className="source js">
               <span
                   className={getClass(entity, 'lupa-entity-name')}
   ```
4. I also hardcoded class names in some places:
   
   ``` javascript
       const name = [
           <span className="storage type function">λ </span>,
           entity.name, '(' , <span className="variable">{params}</span> , ')'];
   ```

So. Basically I've polluted my HTML with Atom specific classes. 

This means that if I want to port plugin to some other environment (e.g. VSCode or browser) I will have to either :
- quick way: redefine styles (i.e. make CSS stylesheet with definition of all CSS classes and colors which I used, e.g.:
  
  ``` css
  .source.js .storage.type.function {
     color: red;
  } 
  ```
- or download theme styles from Atom and redistribute them together with my decoupled plugin 
- or refactor and rethink whole think. Maybe drop some class injectors into JSX etc. (advantages: flexibility, decoupling, code cleanliness. Disadvantages: 1. implementation may take time 2. may be overengineered and not necessary.
